tfhe/src/integer/server_key/radix/shift.rs

```diff
@@ -76,11 +77,10 @@ impl ServerKey {
     /// let dec = cks.decrypt(&ct_res);
     /// assert_eq!(msg >> shift, dec);
     /// ```
-    pub fn unchecked_scalar_right_shift(
-        &self,
-        ct: &RadixCiphertext,
-        shift: u64,
-    ) -> RadixCiphertext {
+    pub fn unchecked_scalar_right_shift<T>(&self, ct: &RadixCiphertext, shift: T) -> RadixCiphertext
+    where
+        u64: CastFrom<T>,
```

Comment:
this works even for u128 because the high bits would be multiples of the bit sizes ?

Reply:
I’m going to be annoying and say that if the objective is to match rust then we need to apply the modulo operator before the u64 conversion 🙂 the bound should not be too bad I believe ?

Reply:
though rust does not have weird non power of two N uN types

Reply:
still feels like we can be consistent without it being too costly so I'd still go for the additional bound I think ?

Reply:
I changed the % to be before the casting

Reply:
yep thanks a lot !